### PR TITLE
mariadb: update build configuration.

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
In release-notes marked deprecated TokuDB plugin.
Added options withStorageRocks and withStorageToku. It is necessary to be able to build MariaDB with jemalloc 5 library. For this, we need to disable TokuDB plugin.
Added zstd compression.

cc @flokli @aanderse @Mic92 
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/sql/mariadb/default.nix [hunk 1]
```diff
@@ -1,9 +1,12 @@
-{ stdenv, fetchurl, fetchFromGitHub, cmake, pkgconfig, makeWrapper, ncurses, zlib, xz, lzo, lz4, bzip2, snappy
+{ stdenv, fetchurl, fetchFromGitHub, cmake, pkgconfig, makeWrapper, ncurses
 , libiconv, openssl, pcre, boost, judy, bison, libxml2, libkrb5, linux-pam, curl
 , libaio, libevent, jemalloc, cracklib, systemd, perl
+, bzip2, lz4, lzo, snappy, xz, zlib, zstd
 , fixDarwinDylibNames, cctools, CoreServices, less
 , numactl # NUMA Support
 , withStorageMroonga ? true, kytea, msgpack, zeromq
+, withStorageRocks ? true
+, withStorageToku ? true, jemalloc450
```
Comment (flokli) [inline pkgs/servers/sql/mariadb/default.nix:9]: please move jemalloc450 in a separate line after `numactl`.

**Review by flokli — COMMENTED** [review]
Apart from a small nit, LGTM.

**Comment by Izorkin** [comment]
Please backport to 20.03.

**Comment by flokli** [comment]
This fails evaluation.

**Comment by Izorkin** [comment]
error - `Failed during the setup of executing nix-env: Os { code: 2, kind: NotFound, message: "No such file or directory" }`

File: nixos/doc/manual/release-notes/rl-2003.xml [hunk 2]
```diff
@@ -757,6 +757,12 @@ auth required pam_succeed_if.so uid >= 1000 quiet
      </warning>
     </para>
    </listitem>
+   <listitem>
+     <para>
+       TokuDB plugin has been disabled in <package>mariadb</package> 10.5. It is recommended to switch
```
Comment (flokli) [inline nixos/doc/manual/release-notes/rl-2003.xml:762]:
```suggestion
       The TokuDB storage engine will be disabled in <package>mariadb</package> 10.5. It is recommended to switch
```

Just to make sure we didn't switch to 10.5 yet, but users are aware to switch to RocksDB.

File: nixos/doc/manual/release-notes/rl-2003.xml [hunk 3]
```diff
@@ -757,6 +757,12 @@ auth required pam_succeed_if.so uid >= 1000 quiet
...
+       to RocksDB. See also <link xlink:href="https://jira.mariadb.org/browse/MDEV-19780">MDEV-19780</link>.
```
Comment (flokli) [inline nixos/doc/manual/release-notes/rl-2003.xml:763]: Maybe link to https://mariadb.com/kb/en/tokudb/ instead.

**Comment by flokli** [comment]
I'm personally fine backporting this to 20.03 - both TokuDB and RocksDB are enabled, and having this in 20.03 (and in the release notes) gives users some time to migrate.

**Comment by Izorkin** [comment]
@GrahamcOfBorg build mariadb.client

**Comment by Izorkin** [comment]
@GrahamcOfBorg build mariadb.client

**Comment by Izorkin** [comment]
@GrahamcOfBorg eval

**Comment by Izorkin** [comment]
@GrahamcOfBorg build mariadb.server

**Comment by Izorkin** [comment]
@GrahamcOfBorg build mariadb.server

**Comment by Izorkin** [comment]
@GrahamcOfBorg build mariadb.client

**Comment by Izorkin** [comment]
@GrahamcOfBorg eval

**Comment by flokli** [comment]
@GrahamcOfBorg test mysql

**Review by flokli — APPROVED** [review]
LGTM. I very much approve bringing this into 20.03, but want wait feedback from @worldofpeace or @disassembler.

**Review by flokli — APPROVED** [review]
LGTM. I very much approve bringing this into 20.03, but want wait feedback from @worldofpeace or @disassembler.

**Review by worldofpeace — APPROVED** [review]
I approve of this being backported.

**Comment by flokli** [comment]
@Izorkin this doesn't seem to apply cleanly on 20.03, even not when I `git cherry-pick -x 4b8d66aa7266ab55f08d53271df36832d9c055d2` before. Can you prepare a PR targetet against 20.03 [cherry-picking these commits](https://nixos.org/nixpkgs/manual/#submitting-changes-stable-release-branches)?

**Pull request merged**